src/react/components/pages/projectSettings/projectSettingsPage.tsx

```diff
@@ -176,29 +186,34 @@ export default class ProjectSettingsPage extends React.Component<IProjectSetting
 
     private onFormSubmit = async (project: IProject) => {
         const isNew = !(!!project.id);
+        try {
+            this.setState({ isCommiting: true });
+            const projectService = new ProjectService();
+            if (!(await projectService.isValidProjectConnection(project))) {
+                return;
+            }
 
-        const projectService = new ProjectService();
-        if (!(await projectService.isValidProjectConnection(project))) {
-            return;
-        }
-
-        if (await this.isValidProjectName(project, isNew)) {
-            toast.error(interpolate(strings.projectSettings.messages.projectExisted, { project }));
-            return;
-        }
+            if (await this.isValidProjectName(project, isNew)) {
+                toast.error(interpolate(strings.projectSettings.messages.projectExisted, { project }));
+                return;
+            }
 
-        await this.deleteOldProjectWhenRenamed(project, isNew);
-        await this.props.applicationActions.ensureSecurityToken(project);
-        await this.props.projectActions.saveProject(project, false, true);
-        removeStorageItem(constants.projectFormTempKey);
+            await this.deleteOldProjectWhenRenamed(project, isNew);
+            await this.props.applicationActions.ensureSecurityToken(project);
+            await this.props.projectActions.saveProject(project, false, true);
+            removeStorageItem(constants.projectFormTempKey);
 
-        toast.success(interpolate(strings.projectSettings.messages.saveSuccess, { project }));
+            toast.success(interpolate(strings.projectSettings.messages.saveSuccess, { project }));
 
-        if (isNew) {
-            this.props.history.push(`/projects/${this.props.project.id}/edit`);
-        } else {
-            this.props.history.goBack();
+            if (isNew) {
+                this.props.history.push(`/projects/${this.props.project.id}/edit`);
+            } else {
+                this.props.history.goBack();
+            }
+        } finally {
+            this.setState({ isCommiting: false });
         }
```

Comment:
Why do we need a try here?

Reply:
there is a try... finally, ensure **_this.setState({ isCommiting: false });_** will execute.